### PR TITLE
Add retry logic to handle priceteller timeouts

### PR DESCRIPTION
**Pull request by rhnonose**
Use `retry` when requesting price suggestions.
We still didn't decide how to handle when the service is not accessible, so I'll return a `Timeout` message for now.

### CODE REVIEW

File: apps/re_integrations/lib/priceteller/price_teller.ex [hunk 1]
```diff
@@ -2,6 +2,8 @@ defmodule ReIntegrations.PriceTeller do
   @moduledoc """
   Context module for price suggestion
   """
+  use Retry
```
Comment (pmargreff) [inline apps/re_integrations/lib/priceteller/price_teller.ex]: this module shouldn't be in our dependencies?

File: apps/re/test/price_suggestions/price_suggestions_test.exs [hunk 2]
```diff
@@ -119,5 +119,26 @@ defmodule Re.PriceSuggestionsTest do
...
+
+    @tag capture_log: true
+    test "should handle timeout error" do
+      %{uuid: uuid} =
+        listing =
+        insert(
+          :listing,
+          rooms: 2,
+          area: 80,
+          address: build(:address, state: "MS", city: "Mah City", street: "Mah Street"),
```
Comment (pmargreff) [inline apps/re/test/price_suggestions/price_suggestions_test.exs]: is this specific info necessary (state, city, and street)? Can we use only `build(:address)` here?

File: apps/re_integrations/lib/priceteller/price_teller.ex (hunk 1 above)
Comment (rhnonose) [reply apps/re_integrations/lib/priceteller/price_teller.ex]: It is in `apps/re_integration/mix.exs`

File: apps/re/test/price_suggestions/price_suggestions_test.exs (hunk 2 above)
Comment (rhnonose) [reply apps/re/test/price_suggestions/price_suggestions_test.exs]: No, probably some copy&pasta duplication 😅

File: apps/re_integrations/lib/priceteller/price_teller.ex (hunk 1 above)
Comment (pmargreff) [reply apps/re_integrations/lib/priceteller/price_teller.ex]: Sorry, missed this.

**Pull request merged**